### PR TITLE
Story3 Get Tasks assigned to Users based on Project ID

### PR DESCRIPTION
**Pull request by CMeier556**
Tasks are correctly filled in each project window and will also return JSON objects based on user_id and project_id in the URL
Date formats correctly
Exceptions are thrown under correct conditions

### CODE REVIEW

File: src/Entities/TaskEntity.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use ProjectManager\Services\DateService;
+
+class TaskEntity implements \JsonSerializable
+{
+    private int $id;
+    private int $project_id;
+    private int $user_id;
+    private string $name;
+    private ?int $estimate;
+    protected ?string $deadline;
```
Comment (mporam) [inline src/Entities/TaskEntity.php]: why is this one different?

File: src/Entities/TaskEntity.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    public function getDeadline(): ?string
```
Comment (mporam) [inline src/Entities/TaskEntity.php]: does this need to be public?

File: src/Services/ConvertToJsonService.php [hunk 3]
```diff
@@ -24,6 +29,16 @@ class ConvertToJsonService
         "data" => []
     ];
 
+    public const INVALID_USER_ID_RESPONSE = [
```
Comment (mporam) [inline src/Services/ConvertToJsonService.php]: why public?

File: src/Services/ConvertToJsonService.php [hunk 4]
```diff
@@ -24,6 +29,16 @@ class ConvertToJsonService
...
+        "message" => "Invalid user ID",
+        "data" => []
+    ];
+
+    public const NO_TASK_ASSIGNED_TO_USER = [
```
Comment (mporam) [inline src/Services/ConvertToJsonService.php]: why public?

File: tests/Services/ConvertToJsonServiceTest.php [hunk 5]
```diff
@@ -46,11 +46,27 @@ public function testConvert_malformed_message()
         \ProjectManager\Services\ConvertToJsonService::convert([], '');
     }
 
+    public function testNoTasksAssignedToUserErrorResponse_success()
+    {
+        $result = \ProjectManager\Services\ConvertToJsonService::NoTasksAssignedToUserErrorResponse();
```
Comment (mporam) [inline tests/Services/ConvertToJsonServiceTest.php]: capital n on the function name? this looks wrong

**Review by mporam — CHANGES_REQUESTED** [review]

File: src/Entities/TaskEntity.php (hunk 1 above)
Comment (CMeier556) [reply src/Entities/TaskEntity.php]: Changed

**Review by mporam — APPROVED** [review]

**Pull request merged**